dipy/core/gradients.py

```diff
@@ -736,3 +736,113 @@ def check_multi_b(gtab, n_bvals, non_zero=True, bmag=None):
         return False
     else:
         return True
+
+
+def btensor_to_bdelta(btens):
+    r"""Compute anisotropy of b-tensor(s)
+
+    Parameters
+    ----------
+    btens : (3, 3) OR (N, 3, 3) numpy.ndarray
+        input b-tensor, or b-tensors, where N = number of b-tensors
+
+    Returns
+    -------
+    bdelta: numpy.ndarray
+        normalized tensor anisotropy(s)
+    bval: numpy.ndarray
+        b-value(s)
+
+    Notes
+    -----
+    Tip: use this to get bdeltas from the GradientTable btens attribute
+    Implementation following [1].
+
+    References
+    ----------
+    .. [1] D. Topgaard, NMR methods for studying microscopic diffusion
+    anisotropy, in: R. Valiullin (Ed.), Diffusion NMR of Confined Systems: Fluid
+    Transport in Porous Solids and Heterogeneous Materials, Royal Society of
+    Chemistry, Cambridge, UK, 2016.
+
+    Examples
+    --------
+    >>> lte = np.array([[1, 0, 0], [0, 0, 0], [0, 0, 0]])
+    >>> bdelta, bval = btensor_to_bdelta(lte)
+    >>> print(f"bdelta={bdelta[0]}; bval={bval[0]}")
+    bdelta=1.0; bval=1.0
+
+    """
+    def _btensor_to_bdelta_2d(btens_2d):
+        """Compute anisotropy of a single b-tensor
+
+        Auxiliary inner function where actual calculation of `bdelta` takes
+        place. The main function `btensor_to_bdelta` then wraps around this to
+        enable support of input (N, 3, 3) arrays, where N = number of b-tensors
+
+        """
+        evals = np.real(np.linalg.eig(btens_2d)[0])
+        bval = np.sum(evals)
+
+        if bval < np.spacing(100):
+            bdelta = 0
+            bval = 0
+            return bdelta, bval
+
+        lambda_iso = (1/3)*bval
+
+        diff_lambdas = np.abs(evals - lambda_iso)
+        evals_zzxxyy = evals[np.argsort(diff_lambdas)[::-1]]
+
+        lambda_zz = evals_zzxxyy[0]
+        lambda_xx = evals_zzxxyy[2]
+        lambda_yy = evals_zzxxyy[1]
+
+        bdelta = (1/(3*lambda_iso))*(lambda_zz-((lambda_yy+lambda_xx)/2))
+
+        tol = np.spacing(1)
+        if np.abs(bval) < tol:
+            bval = 0
+
+        if np.abs(bdelta) < tol:
+            bdelta = 0
+
+        return bdelta, bval
+
+    # Bad input checks
+    value_error_msg = (f"`btens` must be a 2D or 3D numpy array, respectively"
+                       f" with (3, 3) or (N, 3, 3) shape, where N corresponds"
+                       f" to the number of b-tensors")
```

Comment:
f-strings are available from python 3.6.  This is why CI's are failing for the python3.5 matrix. We need to check when they stop the maintenance of python 3.5 and following this info, we either remove this matrix or you will have to make your code compatible with python3.5